minorminer/utils/zephyr/plane_shift.py

```diff
@@ -0,0 +1,163 @@
+    >>> print(f"{ps1 + ps2  = }, {2*ps1 = }")
+    """
+
+    def __init__(
+        self,
+        x_shift: int,
+        y_shift: int,
+    ) -> None:
+        for shift in [x_shift, y_shift]:
+            if not isinstance(shift, int):
+                raise TypeError(f"Expected {shift} to be 'int', got {type(shift)}")
+        if x_shift % 2 != y_shift % 2:
+            raise ValueError(
+                f"Expected x_shift, y_shift to have the same parity, got {x_shift, y_shift}"
+            )
+        self._shift = Shift(x_shift, y_shift)
+
+    @property
+    def x(self) -> int:
+        """Returns the shift in x direction"""
+        return self._shift.x
+
+    @property
+    def y(self) -> int:
+        """Returns the shift in y direction"""
+        return self._shift.y
+
+    def __mul__(self, scale: int | float) -> PlaneShift:
+        """Multiplies the self from left by the number value``scale``.
+
+        Args:
+            scale (int | float): The scale for left-multiplying self with.
+
+        Raises:
+            TypeError: If scale is not 'int' or 'float'.
+            ValueError: If the resulting PlaneShift has non-whole values.
+
+        Returns:
+            PlaneShift: The result of left-multiplying self by scale.
+        """
+        if not isinstance(scale, (int, float)):
+            raise TypeError(f"Expected scale to be int or float, got {type(scale)}")
+        new_shift_x = scale * self._shift.x
+        new_shift_y = scale * self._shift.y
+        if int(new_shift_x) != new_shift_x or int(new_shift_y) != new_shift_y:
+            raise ValueError(f"{scale} cannot be multiplied by {self}")
+        return PlaneShift(int(new_shift_x), int(new_shift_y))
+
+    def __rmul__(self, scale: int | float) -> PlaneShift:
+        """Multiplies the self from right by the number value``scale``.
+
+        Args:
+            scale (int | float): The scale for right-multiplying self with.
+
+        Raises:
+            TypeError: If scale is not 'int' or 'float'.
+            ValueError: If the resulting PlaneShift has non-whole values.
+
+        Returns:
+            PlaneShift: The result of right-multiplying self by scale.
+        """
+        return self * scale
+
+    def __add__(self, other: PlaneShift) -> PlaneShift:
+        """
+        Adds another PlaneShift object to self.
+        Args:
+            other (PlaneShift): The object to add self by.
+
+        Raises:
+            TypeError: If other is not PlaneShift
+
+        Returns:
+            PlaneShift: The displacement in CartesianCoord by self followed by other.
+        """
+        if not isinstance(other, PlaneShift):
+            raise TypeError(f"Expected other to be PlaneShift, got {type(other)}")
+        return PlaneShift(self._shift.x + other._shift.x, self._shift.y + other._shift.y)
+
+    def __iter__(self) -> Iterator[int]:
+        return self._shift.__iter__()
+
+    def __len__(self) -> int:
+        return self._shift.__len__()
```

Comment:
(minor) I'd lean towards using the available functions in these cases.
```suggestion
        return len(self._shift)
```